src/lib.rs

```diff
@@ -576,6 +578,59 @@ fn read_into_buffer(fd: RawFd, buffer: &mut [u8]) -> isize {
     }
 }
 
+/// A RAII guard around a `RawFd` that closes it automatically on drop.
+#[derive(Debug)]
+struct FdGuard {
+    fd: RawFd,
+    close_on_drop: AtomicBool,
+}
+
+impl Deref for FdGuard {
+    type Target = RawFd;
+
+    #[inline]
+    fn deref(&self) -> &Self::Target {
+        &self.fd
+    }
+}
+
+impl Drop for FdGuard {
+    fn drop(&mut self) {
+        if self.close_on_drop.load(Ordering::Relaxed) {
```

Comment:
Can you please explain the reasoning behind `Ordering::Relaxed`? Won't loading the `bool` that way completely negate the guarantees that storing it with `Ordering::SeqCst` would otherwise have?

Reply:
Ooh, I'm really glad you noticed that. I think this is just a typo on my part. This was supposed to be `Ordering::Release` but my editor must have autocompleted the wrong variant.

Thanks so much for catching that, I totally missed it!